validators/cgroup_validator_linux.go

```diff
@@ -142,15 +231,43 @@ func (c *CgroupsValidator) getCgroupV2Subsystems() ([]string, error) {
 	// Some controllers are implicitly enabled by the kernel.
 	// Those controllers do not appear in /sys/fs/cgroup/cgroup.controllers.
 	// https://github.com/torvalds/linux/blob/v5.3/kernel/cgroup/cgroup.c#L433-L434
-	// We assume these are always available, as it is hard to detect availability.
-	// So, we hardcode the following as "pseudo" controllers.
+	// For freezer, we use checkCgroupV2Freeze() to check.
+	// For others, we assume these are always available, as it is hard to detect availability.
+	// We hardcode the following as initial controllers.
 	// - devices: implemented in kernel 4.15
-	// - freezer: implemented in kernel 5.2
-	pseudo := []string{"devices", "freezer"}
+	subsystems := []string{"devices"}
+	freezeSupported, err := checkCgroupV2Freeze()
+	if err != nil {
+		fmt.Printf("error checking cgroups v2 freeze: %v\n", err)
```

Comment:
instead of printing here i think the function should return []string, error, error
```
		subsystems, warn, err = c.getCgroupV2Subsystems()
```

the caller function has it:
```
func (c *CgroupsValidator) Validate(spec SysSpec) (warns, errs []error) {
```